File: app/src/main/java/com/example/phonebook/repos/UsersRepo.java
package com.example.phonebook.repos;

import java.util.Map;
import java.util.NoSuchElementException;

public interface UsersRepo {
    void addUser(String _userLogin, String _userPassword) throws IllegalArgumentException;
    void setSignedUp(Boolean _flag);
    boolean verifyUser(String _user, String _userPassword);
    void setCurrentUser(String _currentUser);
    void deleteUser(String _userLogin) throws NoSuchElementException;
    void updateUser(String _currentUser, String _newUserPassword) throws NoSuchElementException;
    Map <String, ?> getAllUsers();
}

File: app/src/main/java/com/example/phonebook/data/repos/UsersRepo.kt
package com.example.phonebook.data.repos

import com.example.phonebook.data.entities.User

interface UsersRepo {
    fun addUser(user: User): Long
    fun deleteUser(user: User)
    fun updateUser(ID: String, newUser: User)
    fun getUser(login: String): User?
}
